utils/docker-ce/files/dockerd.init

```diff
@@ -9,8 +9,65 @@ json_add_array_string() {
 	json_add_string "" "$1"
 }
 
+docker_bridge_add_config() {
+	local net="$1"
+
+	eval "$(ipcalc.sh "$net")"
+
+	# Add network interface
+	if ! uci -q get network.docker >/dev/null; then
+		uci -q batch >/dev/null <<EOF
+add network interface
+rename network.@interface[-1]=docker
+EOF
+	fi
+
+	uci -q batch >/dev/null <<EOF
+set network.docker.ifname='docker0'
+set network.docker.proto='static'
+set network.docker.netmask='$NETMASK'
+set network.docker.ipaddr='$IP'
+set network.docker.auto='0'
+commit network
+EOF
+	logger -t "dockerd" -p notice "Network interface docker added"
+
+	# Add firewall zone
+	if ! uci -q get firewall.docker >/dev/null; then
+		uci -q batch >/dev/null <<EOF
+add firewall zone
+rename firewall.@zone[-1]=docker
+EOF
+	fi
+
+	uci -q batch >/dev/null <<EOF
+set firewall.docker.network='docker'
+set firewall.docker.input='REJECT'
+set firewall.docker.output='ACCEPT'
+set firewall.docker.forward='REJECT'
+set firewall.docker.name='docker'
+commit firewall
+EOF
+	logger -t "dockerd" -p notice "Firewall zone docker added"
+
+	reload_config
+}
+
+docker_bridge_remove_config() {
```

Comment:
do we want to be able to manually adjust `network.docker` or `firewall.docker` outside of this setup?

Reply:
I have discussed recently with @lisaac about this topic to add and remove docker networks to the network and firewall uci configuration.
The result was, that only the init script should add the default docker0 to the network and firewall configuration.
If we add additional network configuration this is done by luci-app-dockerman and not by the init script.

Reply:
I have also thought about whether we should make the creation of the network and firewall configuration configurable?

After thinking about this I do not like my implementation to delete the configuration and commit theme on service stop. We could introduce a separate init target next to stop/start/restart that would delete the configuration. Maybe clean or purge. The new target should only be executed if f the dockerd service is not running.

Reply:
I'm not sure if I understood all that. 
do you mean the following?
1. we would only add the network to the UCI if it doesn't exist already and just update `network.docker.netmask` and `network.docker.ipaddr` if `bip` changes.
2.  we would only add the firewall to the UCI if it doesn't exist already
3. add a cleanup target that is only manually called to remove the firewall and network

That sounds like a good way to be flexible so anyone can have exactly what they want.

Reply:
Yeah, that's what I meant.
Gonna update my pullrequest.
Thanks for your feedback.